Shared/sdk/SharedUtil.Crypto.h

```diff
@@ -25,18 +26,66 @@ namespace SharedUtil
         SString publicKey, privateKey;
     };
 
-    inline SString Base64encode(const SString& data)
+    inline SString Base64encode(const SString& data, const SString& variant = SString())
     {
-        SString                result;
-        CryptoPP::StringSource ss(data, true, new CryptoPP::Base64Encoder(new CryptoPP::StringSink(result), false));            // Memory is freed automatically
+        SString result;
+
+        if (variant == "URL")
+        {
+            CryptoPP::StringSource ss(data, true, new CryptoPP::Base64URLEncoder(new CryptoPP::StringSink(result), false));
+        }
+        else
+        {
+            CryptoPP::StringSource ss(data, true, new CryptoPP::Base64Encoder(new CryptoPP::StringSink(result), false));
+        }
+
+        return result;
+    }
+
+    inline SString Base64decode(const SString& data, const SString& variant = SString())
+    {
+        std::string result;
```

Comment:
This breaks the NRVO optimization. A type of the returning value shall match the return type of a function. The same for functions below.

Reply:
Done.